redis/asyncio/cluster.py

```diff
@@ -298,6 +306,17 @@ def __init__(
             # Call our on_connect function to configure READONLY mode
             kwargs["redis_connect_func"] = self.on_connect
 
+        if retry or retry_on_error or connection_error_retry_attempts > 0:
+            # Set a retry object for all cluster nodes
+            self.retry = retry or Retry(
+                default_backoff(), connection_error_retry_attempts
+            )
+            if retry_on_error is None:
```

Comment:
If an empty list is passed, it will not get the default value. Better to do: `if not retry_on_error`

Reply:
@utkarshgupta137 is correct. This only checks the None value, but an empty list can be passed, given the type hint.